pkg/catalog/install_test.go

```diff
@@ -0,0 +1,122 @@
+package catalog_test
+
+import (
+	"bytes"
+	"io"
+	"io/ioutil"
+	"net/http"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	kjson "k8s.io/apimachinery/pkg/runtime/serializer/json"
+
+	profilesv1 "github.com/weaveworks/profiles/api/v1alpha1"
+
+	"github.com/weaveworks/pctl/pkg/catalog"
+	"github.com/weaveworks/pctl/pkg/catalog/fakes"
+)
+
+// StringWriter is a test writer to generate output of the subscription into string
+type StringWriter struct {
+	output io.Writer
+}
+
+func (sw *StringWriter) Output(prof *profilesv1.ProfileSubscription) error {
+	e := kjson.NewSerializerWithOptions(kjson.DefaultMetaFactory, nil, nil, kjson.SerializerOptions{Yaml: true, Strict: true})
+	if err := e.Encode(prof, sw.output); err != nil {
+		return err
+	}
+	return nil
+}
+
+var _ = Describe("Install", func() {
```

Comment:
can we have some failure case tests here as well please? 😄

Reply:
Yes, ofc. :D I'm writing it now. :) Just was overeager with sharing. :D